core/python/wrapper/src/core.h

```diff
@@ -34,38 +34,48 @@ class PyMonitoringGenerator : public PyGenerator<T>
 {
 public:
 	using PyGenerator<T>::PyGenerator;
+
 	void onNewSolution(const SolutionBase& s) override { PYBIND11_OVERRIDE_PURE(void, T, onNewSolution, s); }
 };
 
-template <class T = PropagatingEitherWay>
-class PyPropagatingEitherWay : public PyStage<T>
+template <class T = MonitoringGenerator>
+class PubMonitoringGenerator : public PyGenerator<T>
```

Comment:
The pattern to expose protected or private members is slightly different:
https://pybind11.readthedocs.io/en/stable/advanced/classes.html#binding-protected-member-functions

There is no need to derive from PyGenerator<T> here.